armi/physics/fuelCycle/fuelHandlerInterface.py

```diff
@@ -107,13 +112,26 @@ def manageFuel(self, cycle):
             )
             plotting.close()
 
+    def validateLocations(self):
+        """
+        Check that all assemblies have a unique location in the core.
+        """
+        locations = [
+            assembly.getLocation()
+            for assembly in self.r.core.getAssemblies(includeAll=True)
+            if assembly.getLocation()[:3].isnumeric()
+        ]
+
+        if len(locations) != len(set(locations)):
+            raise ValueError("Two or more assemblies share the same core location")
```

Comment:
Should probably output the locations that are problematic here.

Reply:
Updated to output the problematic locations.